File: src/Moryx.FactoryMonitor.Web/src/app/api/models/Moryx/FactoryMonitor/Endpoints/Model/process-model.ts
/* tslint:disable */
/* eslint-disable */
import { ActivityClassification as MoryxControlSystemActivitiesActivityClassification } from '../../../../Moryx/ControlSystem/Activities/activity-classification';
import { ActivityResourceModel as MoryxControlSystemProcessesEndpointsActivityResourceModel } from '../../../../Moryx/ControlSystem/Processes/Endpoints/activity-resource-model';
import { TracingModel as MoryxControlSystemProcessesEndpointsTracingModel } from '../../../../Moryx/ControlSystem/Processes/Endpoints/tracing-model';
import { OrderModel as MoryxFactoryMonitorEndpointsModelOrderModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/order-model';
export interface ProcessModel {
  classification?: MoryxControlSystemActivitiesActivityClassification;
  id?: number;
  instanceId?: null | number;
  isCompleted?: boolean;
  order?: MoryxFactoryMonitorEndpointsModelOrderModel;
  possibleResources?: null | Array<MoryxControlSystemProcessesEndpointsActivityResourceModel>;
  requiredCapabilities?: null | string;
  resource?: MoryxControlSystemProcessesEndpointsActivityResourceModel;
  result?: null | number;
  resultName?: null | string;
  state?: null | string;
  tracing?: MoryxControlSystemProcessesEndpointsTracingModel;
  type?: null | string;
}

File: src/Moryx.FactoryMonitor.Endpoints/Model/ProcessModel.cs
// Copyright (c) 2023, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.ControlSystem.Processes.Endpoints;
using System.Runtime.Serialization;

namespace Moryx.FactoryMonitor.Endpoints.Model
{
    /// <summary>
    /// Extended ProcessActivityModel 
    /// </summary>
    public class ProcessModel : ProcessActivityModel
    {
        [DataMember]
        [Obsolete]
        public virtual OrderModel Order { get; set; }
    }
}

